lib/datadog/tracing/contrib/mysql2/instrumentation.rb

```diff
@@ -57,22 +57,27 @@ def query(sql, options = {})
 
                 Contrib::SpanAttributeSchema.set_peer_service!(span, Ext::PEER_SERVICE_SOURCES)
 
-                propagation_mode = Contrib::Propagation::SqlComment::Mode.new(comment_propagation)
-
-                Contrib::Propagation::SqlComment.annotate!(span, propagation_mode)
-                sql = Contrib::Propagation::SqlComment.prepend_comment(
-                  sql,
-                  span,
-                  trace_op,
-                  propagation_mode
-                )
+                sql = inject_propagation(span, sql, trace_op)
 
                 super(sql, options)
               end
             end
 
             private
 
+            def inject_propagation(span, sql, trace_op)
+              propagation_mode = Contrib::Propagation::SqlComment::Mode.new(datadog_configuration[:comment_propagation])
+
+              Contrib::Propagation::SqlComment.annotate!(span, propagation_mode)
+              Contrib::Propagation::SqlComment.prepend_comment(
+                sql,
+                span,
+                trace_op,
+                propagation_mode,
```

Comment:
Instead of adding another parameter to `prepend_comment`, I suggest refactor and extend `SqlComment::Mode` by injecting the configuration.

```ruby
Contrib::Propagation::SqlComment::Mode.new(datadog_configuration)
```

Reply:
Done!